app/src/main/java/com/example/moviedetailsapplication/network/MovieRepository.kt

```diff
@@ -0,0 +1,28 @@
+package com.example.moviedetailsapplication.network
+
+import com.example.moviedetailsapplication.ui.Movie
+import retrofit2.Call
+
+class MovieRepository(val usersInterface: MoviesInterface) {
+
+    fun getMovies(): Call<APIResponse> {
+        return usersInterface.getMovies()
```

Comment:
we should not return call here
we should enqueue the call and get the response and then convert that into UI model and give that back

Reply:
And see sealed classes to handle this returning response